src/annotator/config/index.js

```diff
@@ -8,7 +10,12 @@ import { toBoolean } from '../../shared/type-coercions';
  *
  * @typedef ConfigDefinition
  * @prop {(settings: SettingsGetters) => any} getValue -
- *   Method to retrieve the value from the incoming source
+ *  Method to retrieve the value from the incoming source
+ * @prop {boolean} allowInBrowserExt -
+ *  Allow this setting to be read in the browser extension. If this is false
+ *  and browser extension context is true, use `defaultValue` if provided otherwise
+ *  ignore the config key
+ * @prop {any} [defaultValue] - Sets a default if `getValue` returns undefined
```

Comment:
`defaultValue` is optional here but you're setting it for every key in the `configDefinitions` map below. _Should_ it be optional?

Reply:
Good Q. Previous to any change changes here any missing value got a "null". I'd like to change this to allow any missing values to be missing (e.g. omitted)

However, I don't feel 100% confident we can do that just yet and it needs a bit more investigation. So the stop gap was to simply fixup the code here to allow missing values to be omitted, but then add a `defaultValue: null` to all the values that don't have some other specialized  default.  In other words, its defined for every setting.

The final step would be to back and see how many of the null defaults we could remove.  Perhaps to make the code more clear, we can make `defaultValue` not be optional for now, and when we do that final audit, it can be reverted to optional?

Reply:
Hi Kyle - This sounds reasonable to me. Can you add a comment in the code that the `defaultValue: null` entries are temporary until you've validated that it can be removed entirely.